Files/Helpers/StorageHelpers.cs

```diff
@@ -11,58 +11,93 @@ namespace Files.Helpers
             {
                 // TODO: In the future, when IStorageItemWithPath will inherit from IStorageItem,
                 // we could implement this code here for getting .lnk files
                 // for now, we can't
                 return default;
             }
-            else if (typeof(IStorageFile).IsAssignableFrom(typeof(TOut)))
-            {
-                await GetFile();
-            }
-            else if (typeof(IStorageFolder).IsAssignableFrom(typeof(TOut)))
+
+            // Fast get attributes
+            bool exists = NativeFileOperationsHelper.GetFileAttributesExFromApp(path, NativeFileOperationsHelper.GET_FILEEX_INFO_LEVELS.GetFileExInfoStandard, out NativeFileOperationsHelper.WIN32_FILE_ATTRIBUTE_DATA itemAttributes);
+            if (exists) // Exists on local storage
             {
-                await GetFolder();
+                // Directory
+                if (itemAttributes.dwFileAttributes.HasFlag(System.IO.FileAttributes.Directory))
+                {
+                    if (typeof(IStorageFile).IsAssignableFrom(typeof(TRequested))) // Wanted file
+                    {
+                        // NotAFile
+                        return default;
+                    }
+                    else // Just get the directory
+                    {
+                        await GetFolder();
+                    }
+                }
+                else // File
+                {
+                    if (typeof(IStorageFolder).IsAssignableFrom(typeof(TRequested))) // Wanted directory
+                    {
+                        // NotAFile
+                        return default;
+                    }
+                    else // Just get the file
+                    {
+                        await GetFile();
+                    }
+                }
             }
-            else if (typeof(IStorageItem).IsAssignableFrom(typeof(TOut)))
+            else // Does not exist or is not present on local storage
             {
-                if (System.IO.Path.HasExtension(path)) // Probably a file
+                if (typeof(IStorageFile).IsAssignableFrom(typeof(TRequested)))
                 {
                     await GetFile();
                 }
-                else // Possibly a folder
+                else if (typeof(IStorageFolder).IsAssignableFrom(typeof(TRequested)))
                 {
                     await GetFolder();
-
-                    if (!folder)
+                }
+                else if (typeof(IStorageItem).IsAssignableFrom(typeof(TRequested)))
+                {
+                    if (System.IO.Path.HasExtension(path)) // Possibly a file
                     {
-                        // It wasn't a folder, so check file then because it wasn't checked
                         await GetFile();
                     }
+                    
+                    if (!file || file.Result == null) // Possibly a folder
```

Comment:
Doesn't this throw NullReferenceException given that it's initialized as `FilesystemResult<BaseStorageFile> file = null;`?

Reply:
I've modified bool operator